#! /bin/sh /usr/share/dpatch/dpatch-run
## 23_check_smb_password.dpatch by Alexander Wirt <formorer@debian.org>
##
## DP: If there is no password set call smbclient with -N (suppress password prompt, bugs.debian.org/425129)

@DPATCH@
diff -urNad nagios-plugins-1.4.11~/plugins-scripts/check_disk_smb.pl nagios-plugins-1.4.11/plugins-scripts/check_disk_smb.pl
--- nagios-plugins-1.4.11~/plugins-scripts/check_disk_smb.pl	2008-05-02 07:13:08.000000000 +0200
+++ nagios-plugins-1.4.11/plugins-scripts/check_disk_smb.pl	2008-05-02 07:15:23.000000000 +0200
@@ -79,6 +79,7 @@
 
 ($opt_p) || ($opt_p = shift) || ($opt_p = "");
 my $pass = $1 if ($opt_p =~ /(.*)/);
+$pass = "-N" if ($opt_p eq "");
 
 ($opt_w) || ($opt_w = shift) || ($opt_w = 85);
 my $warn = $1 if ($opt_w =~ /^([0-9]{1,2}\%?|100\%?|[0-9]+[kMG])$/);
